server/data/statementsClient.ts

```diff
@@ -2,13 +2,15 @@
 import format from 'pg-format'
 import type { QueryPerformer } from './dataAccess/db'
 import { StatementStatus } from '../config/types'
+import { PageResponse, buildPageResponse, HasTotalCount, offsetAndLimitForPage } from '../utils/page'
 
 export default class StatementsClient {
   constructor(private readonly query: QueryPerformer) {}
 
-  async getStatements(userId: string): Promise<StatementSummary[]> {
-    const result = await this.query<StatementSummary>({
-      text: `select r.id
+  async getStatements(userId: string, page: number): Promise<PageResponse<StatementSummary>> {
+    const [offset, limit] = offsetAndLimitForPage(page)
+    const result = await this.query<HasTotalCount<StatementSummary>>({
+      text: `select r.id, count(*) OVER() AS "totalCount"
```

Comment:
using a window function to calculate total count. This will add the total count to each row.

page is 1-indexed integer passed directly from the frontend